### PR TITLE
Updating Bayesian Neural Network Learning Details 

### PR DESCRIPTION
**Pull request by Aakanksha-Rana**
Adding:
- Variational Meshnet (No MC flag needed for 'layers' as they have inbuilt estimators!). Check comments for more specific details.
- Options for BatchNorm/Group in there.
- Loss combinations (Note: Dice is imp for Binary segmentation task!)
- Prior settings.  
- Semi-bayesian Unet

### CODE REVIEW

**Comment by Hoda1394** [comment]
Thanks, @Aakanksha-Rana I will check and merge them. probably, I need to update some other files based on these PRs.

@satra- FYI

**Comment by Aakanksha-Rana** [comment]
@Hoda1394 
There are couple of ways to warm start. One way is to run with likelihood loss (or other loss we discussed) for some epochs... save its weights and then rerun by loading old saved weights with KLD scaled with scaled factor... 
I tried some more adaptive ways with either checkpoints or one method below which might be more useful as you are using train_on_batch feature  
we can simply add the scale variable in the loop Like this.

```
Initial_epoch = 0 ; scaling_start_epoch = 5; scaling_increase_per_epoch = 0.3  # vars 
with strategy.scope():
    model = model(...) #initialize models 
    if initial_epoch >= scaling_start_epoch:
        scale_factor = (min(1., scale_factor + (initial_epoch - scaling_start_epoch) * scaling_increase_per_epoch))*steps_per_epoch
    scale_factor= tf.keras.variable(scale_factor)
    model.compile(....)      
        # training loop
    for epoch in range(epochs):
        for steps, (inputs,labels) in enumerate(dataset_train.take(500)):
            error, acc= model.train_on_batch(inputs,labels)
        if epoch >= scale_start_epoch:
            new_scaling_factor = same as aobe
            K.set_value(kl_alpha, new_kl_alpha)
```

Let me know if you use model.fit, only then callback is useful.

**Comment by Aakanksha-Rana** [comment]
adding bayesian utils
